### PR TITLE
Feature/docker io main thread

### PR DESCRIPTION
**Pull request by rafaelhbarros**

### CODE REVIEW

File: binstar_build_client/worker/docker_worker.py [hunk 1]
```diff
@@ -14,9 +17,30 @@
...
+
+    def wait(self):
+        return self.cli.wait(self.cont)
+
+    def remove(self):
+        self.cli.remove_container(self.cont, v=True)
+
+    def readline(self):
+        try:
+            return self.stream.next()
```
Comment (srossross) [inline binstar_build_client/worker/docker_worker.py]: I think this should be `next(self.stream)`

Comment (rafaelhbarros) [inline binstar_build_client/worker/docker_worker.py]: I will change to reflect that.

File: binstar_build_client/worker/worker.py [hunk 2]
```diff
@@ -61,6 +61,13 @@ def remove_files_after(files):
                 os.unlink(filename)
 
 
+class BuildProcess(sp.Popen):
+    def kill(self):
+        kill_tree(self)
+
+    def readlines(self):
+        return self.stdout.readlines()
```
Comment (srossross) [inline binstar_build_client/worker/worker.py]: @rafaelhbarros does this work?

Comment (srossross) [inline binstar_build_client/worker/worker.py]: I tried this, for python2  `readlines` returns a list. and xreadlines just hangs.

Comment (rafaelhbarros) [inline binstar_build_client/worker/worker.py]: that's a typo. It should be `readline`.

Comment (rafaelhbarros) [inline binstar_build_client/worker/worker.py]: I have also moved that to a different file

**Pull request closed without merging**